src/main/java/seedu/address/logic/commands/AddCommand.java

```diff
@@ -37,13 +42,23 @@ public class AddCommand extends Command {
     public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
 
     private final Person toAdd;
+    private final Set<String> duplicateFields;
 
     /**
-     * Creates an AddCommand to add the specified {@code Person}
+     * Creates an AddCommand to add the specified {@code Person} with duplicated fields {@code duplicateFields}.
      */
-    public AddCommand(Person person) {
+    public AddCommand(Person person, Set<Prefix> duplicateFields) {
```

Comment:
I don't think this is a good approach. AddCommand should not need to know about duplicate fields.